src/main/java/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -139,8 +139,21 @@ public void writeExternal(DataOutputStream out) throws IOException {
     @Override
     public DataInstance initialize(InstanceInitializationFactory initializer, String instanceId) {
         base = new InstanceBase(instanceId);
-        root = initializer.generateRoot(this);
-        base.setChild(root);
+        InstanceRoot instanceRoot = initializer.generateRoot(this);
+        // this indirectly calls `this.copyFromSource` via the InstanceRoot so that we call the
+        // correct method based on the type
+        instanceRoot.setupNewCopy(this);
```

Comment:
I am not able to grasp how the `InstanceRoot` is helping persisiting `useCaseTemplate` .  `setupNewCopy` only seems to be called from here and should result in executing the block - 

````
 root = instanceRoot.getRoot();
 base.setChild(root);
````

How is it then the `useCaseTemplate` getting copied here ?

Reply:
If the `instanceRoot` is an `ExternalDataInstanceSource` then it will call also copy the `source` value:

```
     public void copyFromSource(ExternalDataInstanceSource source) {
        //parent copy
        copyFromSource((InstanceRoot)source);
        this.source = source;
    }
```
https://github.com/dimagi/commcare-core/pull/1053/files/095d42695f3ecab008366af0294ca10b3a6ed084#diff-d872b3df15633847b778eaa24e6797f9a6339f68c9981ac04ed16d9eb5ade1c4R154-R158

Reply:
this makes sense, thanks!